tests/integration/test_charm.py

```diff
@@ -198,3 +201,123 @@ async def test_scale_down_capablities(ops_test: OpsTest) -> None:
 
     assert set(member_ips) == set(ip_addresses), "mongod config contains deleted units"
+
+
+async def test_replication_across_members(ops_test: OpsTest) -> None:
+    """Check consistency, ie write to primary, read data from secondaries."""
+    # first find primary, write to primary, then read from each unit
+    ip_addresses = [unit.public_address for unit in ops_test.model.applications[APP_NAME].units]
+    primary = await replica_set_primary(ip_addresses, ops_test)
+    password = await get_password(ops_test)
+    client = MongoClient(unit_uri(primary, password), directConnection=True)
+
+    db = client["new-db"]
+    test_collection = db["test_collection"]
+    ubuntu = {"release_name": "Focal Fossa", "version": 20.04, "LTS": True}
+    test_collection.insert(ubuntu)
+
+    client.close()
+
+    secondaries = set(ip_addresses) - set([primary])
+    logger.error(secondaries)
+    for secondary in secondaries:
+        client = MongoClient(unit_uri(secondary, password), directConnection=True)
+
+        db = client["new-db"]
+        test_collection = db["test_collection"]
+        query = test_collection.find({}, {"release_name": 1})
+        logger.error(query[0]["release_name"])
+        assert query[0]["release_name"] == "Focal Fossa"
+
+        client.close()
+
+
+async def test_unique_cluster_dbs(ops_test: OpsTest) -> None:
+    """Verify unique clusters do not share DBs."""
+    # deploy new cluster
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(my_charm, num_units=1, application_name=ANOTHER_DATABASE_APP_NAME)
+    await ops_test.model.wait_for_idle()
+
+    # write data to new cluster
+    ip_addresses = [
+        unit.public_address
+        for unit in ops_test.model.applications[ANOTHER_DATABASE_APP_NAME].units
+    ]
+    password = await get_password(ops_test, app=ANOTHER_DATABASE_APP_NAME)
+    client = replica_set_client(ip_addresses, password, app=ANOTHER_DATABASE_APP_NAME)
+
+    db = client["new-db"]
+    test_collection = db["test_collection"]
+    ubuntu = {"release_name": "Jammy Jelly", "version": 22.04, "LTS": False}
+    test_collection.insert(ubuntu)
+
+    # read all entries from new cluster
+    cursor = test_collection.find({})
+    cluster_1_entries = set()
+    for document in cursor:
+        logger.error(document)
+        cluster_1_entries.add(document["release_name"])
+
+    client.close()
+
+    # read all entries from other cluster
+    ip_addresses = [unit.public_address for unit in ops_test.model.applications[APP_NAME].units]
+    password = await get_password(ops_test)
+    client = replica_set_client(ip_addresses, password)
+
+    db = client["new-db"]
+    test_collection = db["test_collection"]
+
+    # read all entries from original cluster
+    cursor = test_collection.find({})
+    cluster_2_entries = set()
+    for document in cursor:
+        cluster_2_entries.add(document["release_name"])
+
+    client.close()
```

Comment:
Normally I wouldn't bother DRYing tests, but this is a fair whack of code that's duplicated and distracts from the intent of the test. Suggest putting this into a function and having:

```python
entries_1 = blah(1)
entries_2 = blah(2)
```